platforms/x86-64/core/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+#![no_std]
+extern crate alloc;
+
+use core::time::Duration;
+use hal_core::{boot::BootInfo, PAddr};
+use hal_x86_64::cpu::local::GsLocalData;
+pub use hal_x86_64::{
+    cpu::{local::LocalKey, wait_for_interrupt},
+    mm,
+};
+use kernel::{mnemos_alloc::containers::Box, Kernel, KernelSettings};
+
+pub mod acpi;
+pub mod interrupt;
+
+pub fn init(bootinfo: &impl BootInfo, rsdp_addr: Option<PAddr>) -> &'static Kernel {
+    // TODO: init early tracing?
+    interrupt::enable_exceptions();
+    bootinfo.init_paging();
+
+    // TODO: init allocator!
+
+    // TODO: PCI?
+
+    init_acpi(bootinfo, rsdp_addr);
+
+    // init boot processor's core-local data
+    unsafe {
+        GsLocalData::init();
+    }
+    tracing::info!("set up the boot processor's local data");
+
+    let k = {
+        let settings = KernelSettings {
+            max_drivers: 64, // we are a big x86 system with lots of RAM, this can probably be an even bigger number!
+            timer_granularity: Duration::from_millis(10),
+        };
+
+        unsafe {
+            Box::into_raw(Kernel::new(settings).expect("cannot initialize kernel"))
+                .as_ref()
+                .unwrap()
+        }
+    };
+
+    // TODO: spawn drivers (UART, keyboard, ...)
+    k
+}
+
+pub fn run(bootinfo: &impl BootInfo, k: &'static Kernel) -> ! {
```

Comment:
this will probably also need to take some kind of "CPU core context" eventually when we actually get to SMP support...